### ISSUE TITLE
Map hangs with JNI Error: weak global ref table overflow 

### ISSUE THREAD

**Issue by asheemmamoowala** [mapbox/mapbox-gl-native#10301]
Using the test app on a Galaxy S8 device, I'm able to reproduce the hang quite easily

### Steps to trigger behavior

 1. Use the Simple Map demo of the test app
 2. Pitch the map
 3. Zoom to z10+
 4. Pan the map repeatedly

### Expected behavior
Map continues to pan smoothly

### Actual behavior
After about 45s to 1m of panning the map just hangs. The app itself also becomes unresponsive.

Android Monitor reports:
```
JNI ERROR (app bug): weak global reference table overflow (max=51200)
weak global reference table dump:
 Last 10 entries (of 51200):
    51199: 0x12da0c10 java.lang.Thread
    51198: 0x12d926e0 com.mapbox.mapboxsdk.maps.renderer.MapRendererRunnable
    51197: 0x12da0b80 java.lang.Thread
    51196: 0x12d92680 com.mapbox.mapboxsdk.maps.renderer.MapRendererRunnable
    51195: 0x12da0af0 java.lang.Thread
    51194: 0x12d92630 com.mapbox.mapboxsdk.maps.renderer.MapRendererRunnable
    51193: 0x12da0a60 java.lang.Thread
    51192: 0x12d925e0 com.mapbox.mapboxsdk.maps.renderer.MapRendererRunnable
    51191: 0x12da0940 java.lang.Thread
    51190: 0x12d92580 com.mapbox.mapboxsdk.maps.renderer.MapRendererRunnable
```

[Monitor Log](https://github.com/mapbox/mapbox-gl-native/files/1420281/weakreferror.txt)

**Comment by ivovandongen** [comment]
@asheemmamoowala I'm trying to reproduce this on master (assuming this is the branch to check btw). Doesn't seem to reproduce on the Pixel.


Added the following code to pan around. Could you check if this triggers the issue for you?

```
diff --git a/platform/android/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/activity/maplayout/SimpleMapActivity.java b/platform/android/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/activity/maplayout/SimpleMapActivity.java
index 8f8a5af3c..372b364d1 100644
--- a/platform/android/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/activity/maplayout/SimpleMapActivity.java
+++ b/platform/android/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/activity/maplayout/SimpleMapActivity.java
@@ -3,7 +3,12 @@ package com.mapbox.mapboxsdk.testapp.activity.maplayout;
 import android.os.Bundle;
 import android.support.v7.app.AppCompatActivity;
 
+import com.mapbox.mapboxsdk.camera.CameraPosition;
+import com.mapbox.mapboxsdk.camera.CameraUpdateFactory;
+import com.mapbox.mapboxsdk.geometry.LatLng;
 import com.mapbox.mapboxsdk.maps.MapView;
+import com.mapbox.mapboxsdk.maps.MapboxMap;
+import com.mapbox.mapboxsdk.maps.OnMapReadyCallback;
 import com.mapbox.mapboxsdk.testapp.R;
 
 /**
@@ -20,6 +25,28 @@ public class SimpleMapActivity extends AppCompatActivity {
 
     mapView = (MapView) findViewById(R.id.mapView);
     mapView.onCreate(savedInstanceState);
+
+    mapView.getMapAsync(new OnMapReadyCallback() {
+      @Override
+      public void onMapReady(MapboxMap mapboxMap) {
+        mapboxMap.setCameraPosition(new CameraPosition.Builder().target(new LatLng()).zoom(10).tilt(60).build());
+        animate(mapboxMap);
+      }
+    });
+  }
+
+  private void animate(final MapboxMap mapboxMap) {
+    mapboxMap.animateCamera(CameraUpdateFactory.scrollBy(200, 0), 20, new MapboxMap.CancelableCallback() {
+      @Override
+      public void onCancel() {
+
+      }
+
+      @Override
+      public void onFinish() {
+        animate(mapboxMap);
+      }
+    });
   }
 
   @Override
diff --git a/platform/android/dependencies.gradle b/platform/android/dependencies.gradle
index 521084559..44891584c 100644
--- a/platform/android/dependencies.gradle
+++ b/platform/android/dependencies.gradle
@@ -42,7 +42,7 @@ ext {
 
             // square crew
             timber                 : 'com.jakewharton.timber:timber:4.5.1',
-            okhttp3                : 'com.squareup.okhttp3:okhttp:3.9.0',
+            okhttp3                : 'com.squareup.okhttp3:okhttp:3.10.0-SNAPSHOT',
             leakCanaryDebug        : "com.squareup.leakcanary:leakcanary-android:${leakCanaryVersion}",
             leakCanaryRelease      : "com.squareup.leakcanary:leakcanary-android-no-op:${leakCanaryVersion}"
     ]
```

**Comment by ivovandongen** [comment]
Tried some variations of this on my Galaxy Nexus as well. No crashes yet.

**Comment by ivovandongen** [comment]
Adding some manual counting on the global weak references shows that we are cleaning them up properly. The count reaches 100-200 at some points, but returns to 0 after GC on my devices:

```
diff --git a/platform/android/src/jni/generic_global_ref_deleter.hpp b/platform/android/src/jni/generic_global_ref_deleter.hpp
index 7239e361a..4d3b2d206 100644
--- a/platform/android/src/jni/generic_global_ref_deleter.hpp
+++ b/platform/android/src/jni/generic_global_ref_deleter.hpp
@@ -1,9 +1,17 @@
 #pragma once
 
+#include <mbgl/util/logging.hpp>
+
 #include <jni/jni.hpp>
 
 #include "../attach_env.hpp"
 
+#include <atomic>
+
+namespace {
+    static std::atomic<int> WEAK_REF_COUNT;
+}
+
 namespace mbgl {
 namespace android {
 
@@ -35,8 +43,12 @@ public:
     {
         if (p)
         {
+            int previous = WEAK_REF_COUNT.fetch_add(-1);
+            Log::Info(Event::JNI, "Deleting weak ref. Current count: %i", previous -1);
             auto env = AttachEnv();
             env->DeleteWeakGlobalRef(jni::Unwrap(p->Get()));
+        } else {
+            Log::Error(Event::JNI, "Cannot delete weak ref");
         }
     }
 };
@@ -44,6 +56,8 @@ public:
 template < class TagType >
 GenericUniqueWeakObject<TagType> SeizeGenericWeakRef(JNIEnv&, jni::Object<TagType>&& object)
 {
+    int previous = WEAK_REF_COUNT.fetch_add(1);
+    Log::Info(Event::JNI, "Creating weak ref. Current count:  %i", previous + 1);
     return GenericUniqueWeakObject<TagType>(jni::PointerToValue<jni::Object<TagType>>(std::move(object)), GenericWeakObjectRefDeleter<TagType>());
 };
 
```

**Comment by ivovandongen** [comment]
@asheemmamoowala Is there more to the log file you originally posted? Adding up the numbers in the log file roughly, I don't get up to 51200.

The summary indicates `MapRendererRunnable`, but the log shows that there are "only" 616 instances.

**Comment by asheemmamoowala** [comment]
I got the log using verbose logging in Android Monitor. The rest of the log is from OkHttp.
It's not crashing today with or without the patches.

**Comment by tobrun** [comment]
Capturing from @danesfeder and from https://github.com/mapbox/mapbox-navigation-android/issues/451 that this seem to impact other devices than solely the Samsung Galaxy s8.

**Comment by Grsmto** [comment]
Hi guys,
Any idea on a workaround for this on a potential way to avoid the crash?
Only thing I see at the moment is rolling back to a previous version of the SDK.
If I can provide anything to help debugging this please let me know :)

Thanks!

**Comment by ivovandongen** [comment]
@Grsmto If you are in a position to retest with #10415, that would be great. Haven't been able to reproduce myself.

**Comment by Grsmto** [comment]
@ivovandongen hey, sorry I tried to test but it's quite hard because it seems to be happening in specific circumstances (from my tests, it happens when you pan the map and trigger a camera ease at the same time). I couldn't test on this scenario as this would mean that I use your branch as a dependency of my project and I was unsuccessful (I tried using Jitpack...).
I saw you guys merged so hopefully I'll be able to test with the SNAPSHOT build! :)

**Comment by ivovandongen** [comment]
@Grsmto Thanks for trying, and yes, if you could try with the next snapshot that would be great!

**Comment by Nexxado** [comment]
I'm still seeing this on version 5.3.2
I'm following the user's location by easing the camera on each location update.

Log:
```
W/art: Attempt to remove non-JNI local reference, dumping thread
W/art: weak global reference table dump:
W/art:   Last 10 entries (of 50800):
W/art:     50799: 0x12d2e280 java.lang.Thread
W/art:     50798: 0x12d2e1f0 java.lang.Thread
W/art:     50797: 0x12d2e160 java.lang.Thread
W/art:     50796: 0x12d2e0d0 java.lang.Thread
W/art:     50795: 0x12cfaf70 java.lang.Thread
W/art:     50794: 0x12cfaee0 java.lang.Thread
W/art:     50793: 0x12cfae50 java.lang.Thread
W/art:     50792: 0x12cfad30 java.lang.Thread
W/art:     50791: 0x12cfaca0 java.lang.Thread
W/art:     50790: 0x12cfac10 java.lang.Thread
W/art:   Summary:
W/art:         1 of com.android.tools.profiler.support.memory.VmStatsSampler
W/art:         1 of java.util.TimerThread
W/art:         1 of com.mapbox.mapboxsdk.maps.NativeMapView
W/art:         1 of android.opengl.GLSurfaceView$GLThread
W/art:         1 of com.mapbox.mapboxsdk.maps.MapView$6
W/art:         1 of android.opengl.GLSurfaceView
W/art:       203 of java.lang.Thread (203 unique instances)
W/art:      5595 of java.lang.Class (5595 unique instances)
W/art:         1 of byte[] (1 elements)
W/art:         3 of byte[] (1440 elements) (3 unique instances)
W/art:         3 of byte[] (5016 elements) (3 unique instances)
W/art:         1 of byte[] (6912 elements)
W/art:         5 of byte[] (9216 elements) (5 unique instances)
W/art:         4 of byte[] (10080 elements) (4 unique instances)
W/art:         1 of byte[] (14880 elements)
W/art:         1 of byte[] (17280 elements)
W/art:         9 of byte[] (20736 elements) (9 unique instances)
W/art:        27 of byte[] (26244 elements) (27 unique instances)
W/art:         1 of byte[] (32256 elements)
W/art:         1 of byte[] (32448 elements)
W/art:         2 of byte[] (34848 elements) (2 unique instances)
W/art:         2 of byte[] (35344 elements) (2 unique instances)
W/art:        19 of byte[] (36864 elements) (19 unique instances)
W/art:         3 of byte[] (46656 elements) (3 unique instances)
W/art:         2 of byte[] (47808 elements) (2 unique instances)
W/art:         1 of byte[] (58176 elements)
W/art:         1 of byte[] (64260 elements)
W/art:         3 of byte[] (79488 elements) (3 unique instances)
W/art:         1 of byte[] (82368 elements)
W/art:         4 of byte[] (82944 elements) (4 unique instances)
W/art:         4 of byte[] (108000 elements) (4 unique instances)
W/art:        10 of byte[] (140160 elements) (10 unique instances)
W/art:         5 of byte[] (270400 elements) (5 unique instances)
W/art:         1 of java.lang.ThreadGroup
W/art:        31 of java.lang.DexCache (31 unique instances)
W/art:         4 of dalvik.system.PathClassLoader (1 unique instances)
W/art:         1 of android.net.ConnectivityThread
W/art:         1 of android.os.HandlerThread
W/art:        82 of android.view.RenderNode (82 unique instances)
I/art: HeapInfo: checking storage, bytes_hprof_total: 0, bytes_available: 14192148480, bytes_fileMinNeeded: 615186064
I/art: HeapInfo: bytes_available: 14192148480, bytes_fileMinNeeded: 615186064
E/art: thread suspend timeout: "Thread-44947", sysTid=12656
E/art: thread suspend timeout: "Thread-44949", sysTid=12659
E/art: thread suspend timeout: "Thread-44950", sysTid=12658
E/art: Unexpected time out during suspend all.
E/art: Unexpected time out during suspend all.: Invalid argument
I/art: NativeCrashSigHandler RECEIVE SIGNAL: 6, PID: 12458, TID: 12657
I/art: VMHOOK: rlim_cur : 18446744073709551615 pid:12458
A/libc: Fatal signal 6 (SIGABRT), code -6 in tid 12657 (Thread-44948)

        [ 01-28 19:50:53.566  4133: 4133 W/         ]
        debuggerd: handling request: pid=12458 uid=10213 gid=10213 tid=12657
Disconnected from the target VM, address: 'localhost:8600', transport: 'socket'
```

**Comment by gonzalt03** [comment]
With `com.mapbox.mapboxsdk:mapbox-android-navigation-ui:0.9.0`, same problem. Freeze and this log :

```
A/art: art/runtime/indirect_reference_table.cc:128] JNI ERROR (app bug): weak global reference table overflow (max=51200)
A/art: art/runtime/indirect_reference_table.cc:128] weak global reference table dump:
A/art: art/runtime/indirect_reference_table.cc:128]   Last 10 entries (of 51200):
A/art: art/runtime/indirect_reference_table.cc:128]     51199: 0x12e889d0 java.lang.Thread
A/art: art/runtime/indirect_reference_table.cc:128]     51198: 0x12e88820 java.lang.Thread
A/art: art/runtime/indirect_reference_table.cc:128]     51197: 0x12e88790 java.lang.Thread
A/art: art/runtime/indirect_reference_table.cc:128]     51196: 0x12e88700 java.lang.Thread
A/art: art/runtime/indirect_reference_table.cc:128]     51195: 0x12e88670 java.lang.Thread
A/art: art/runtime/indirect_reference_table.cc:128]     51194: 0x12e885e0 java.lang.Thread
A/art: art/runtime/indirect_reference_table.cc:128]     51193: 0x12e88550 java.lang.Thread
A/art: art/runtime/indirect_reference_table.cc:128]     51192: 0x12e884c0 java.lang.Thread
A/art: art/runtime/indirect_reference_table.cc:128]     51191: 0x12e88430 java.lang.Thread
A/art: art/runtime/indirect_reference_table.cc:128]     51190: 0x12e883a0 java.lang.Thread
A/art: art/runtime/indirect_reference_table.cc:128]   Summary:
A/art: art/runtime/indirect_reference_table.cc:128]         1 of com.google.android.gms.measurement.internal.zzw$zzd
A/art: art/runtime/indirect_reference_table.cc:128]         1 of com.facebook.stetho.server.ServerManager$1
A/art: art/runtime/indirect_reference_table.cc:128]         1 of com.google.android.gms.internal.zzqc$zzb
A/art: art/runtime/indirect_reference_table.cc:128]         3 of csq (1 unique instances)
A/art: art/runtime/indirect_reference_table.cc:128]         1 of com.mapbox.services.android.navigation.v5.navigation.NavigationEngine
A/art: art/runtime/indirect_reference_table.cc:128]         1 of java.util.TimerThread
A/art: art/runtime/indirect_reference_table.cc:128]         1 of okio.AsyncTimeout$Watchdog
A/art: art/runtime/indirect_reference_table.cc:128]         1 of com.mapbox.mapboxsdk.maps.NativeMapView
A/art: art/runtime/indirect_reference_table.cc:128]         2 of android.opengl.GLSurfaceView (2 unique instances)
A/art: art/runtime/indirect_reference_table.cc:128]         1 of com.squareup.picasso.Dispatcher$DispatcherThread
A/art: art/runtime/indirect_reference_table.cc:128]         1 of com.squareup.picasso.Picasso$CleanupThread
A/art: art/runtime/indirect_reference_table.cc:128]         2 of com.mapbox.mapboxsdk.maps.MapView$6 (2 unique instances)
A/art: art/runtime/indirect_reference_table.cc:128]         2 of android.opengl.GLSurfaceView$GLThread (2 unique instances)
A/art: art/runtime/indirect_reference_table.cc:128]       262 of java.lang.Thread (262 unique instances)
A/art: art/runtime/indirect_reference_table.cc:128]      7494 of java.lang.Class (7494 unique instances)
A/art: art/runtime/indirect_reference_table.cc:128]         1 of byte[] (1 elements)
A/art: art/runtime/indirect_reference_table.cc:128]         3 of byte[] (576 elements) (3 unique instances)
A/art: art/runtime/indirect_reference_table.cc:128]         1 of byte[] (1024 elements)
A/art: art/runtime/indirect_reference_table.cc:128]         3 of byte[] (2200 elements) (3 unique instances)
A/art: art/runtime/indirect_reference_table.cc:128]         1 of byte[] (3072 elements)
A/art: art/runtime/indirect_reference_table.cc:128]         5 of byte[] (4096 elements) (5 unique instances)
A/art: art/runtime/indirect_reference_table.cc:128]         4 of byte[] (4480 elements) (4 unique instances)
A/art: art/runtime/indirect_reference_table.cc:128]         1 of byte[] (6560 elements)
A/art: art/runtime/indirect_reference_table.cc:128]         1 of byte[] (7680 elements)
A/art: art/runtime/indirect_reference_table.cc:128]        19 of byte[] (9216 elements) (19 unique instances)
A/art: art/runtime/indirect_reference_table.cc:128]        27 of byte[] (11664 elements) (27 unique instances)
A/art: art/runtime/indirect_reference_table.cc:128]         1 of byte[] (12544 elements)
A/art: art/runtime/indirect_reference_table.cc:128]         1 of byte[] (14336 elements)
A/art: art/runtime/indirect_reference_table.cc:128]         1 of byte[] (14352 elements)
A/art: art/runtime/indirect_reference_table.cc:128]         2 of byte[] (15376 elements) (2 unique instances)
A/art: art/runtime/indirect_reference_table.cc:128]         2 of byte[] (15488 elements) (2 unique instances)
A/art: art/runtime/indirect_reference_table.cc:128]        21 of byte[] (16384 elements) (21 unique instances)
A/art: art/runtime/indirect_reference_table.cc:128]         3 of byte[] (20736 elements) (3 unique instances)
A/art: art/runtime/indirect_reference_table.cc:128]         2 of byte[] (21120 elements) (2 unique instances)
A/art: art/runtime/indirect_reference_table.cc:128]         2 of byte[] (25600 elements) (2 unique instances)
A/art: art/runtime/indirect_reference_table.cc:128]         1 of byte[] (28560 elements)
A/art: art/runtime/indirect_reference_table.cc:128]         1 of byte[] (34272 elements)
A/art: art/runtime/indirect_reference_table.cc:128]         3 of byte[] (35328 elements) (3 unique instances)
A/art: art/runtime/indirect_reference_table.cc:128]         1 of byte[] (36608 elements)
A/art: art/runtime/indirect_reference_table.cc:128]        23 of byte[] (36864 elements) (23 unique instances)
art/runtime/indirect_reference_table.cc:128]         4 of byte[] (49600 elements) (4 unique instances)
art/runtime/indirect_reference_table.cc:128]         1 of byte[] (52104 elements)
art/runtime/indirect_reference_table.cc:128]         1 of byte[] (53760 elements)
art/runtime/indirect_reference_table.cc:128]        10 of byte[] (62080 elements) (10 unique instances)
art/runtime/indirect_reference_table.cc:128]         1 of byte[] (65536 elements)
art/runtime/indirect_reference_table.cc:128]         1 of byte[] (66080 elements)
art/runtime/indirect_reference_table.cc:128]         5 of byte[] (119716 elements) (5 unique instances)
art/runtime/indirect_reference_table.cc:128]         2 of byte[] (640000 elements) (2 unique instances)
art/runtime/indirect_reference_table.cc:128]         1 of byte[] (1048576 elements)
art/runtime/indirect_reference_table.cc:128]         1 of byte[] (2560000 elements)
art/runtime/indirect_reference_table.cc:128]         1 of java.lang.ThreadGroup
art/runtime/indirect_reference_table.cc:128]        96 of java.lang.DexCache (96 unique instances)
art/runtime/indirect_reference_table.cc:128]         6 of dalvik.system.PathClassLoader (3 unique instances)
art/runtime/indirect_reference_table.cc:128]         1 of android.view.SurfaceView
art/runtime/indirect_reference_table.cc:128]         1 of android.net.ConnectivityThread
art/runtime/indirect_reference_table.cc:128]         3 of android.os.HandlerThread (3 unique instances)
art/runtime/indirect_reference_table.cc:128]       546 of android.view.RenderNode (546 unique instances)
art/runtime/indirect_reference_table.cc:128] 
```

**Comment by tobrun** [comment]
With the upcoming 6.0.0 release we added multiple missing DeleteLocalRef calls. Feel free to comment back if this is still reproducible with 6.0.0-beta.3.

**Comment by ugochirico** [comment]
I'm using 6.4.0 and on my OnePlus 6 I got the same error:

`09-13 16:45:22.359 17010-19397/com.andreucci.andreuccicodrive A/zygote64: indirect_reference_table.cc:256] JNI ERROR (app bug): weak global reference table overflow (max=51200)
    indirect_reference_table.cc:256] weak global reference table dump:
    indirect_reference_table.cc:256]   Last 10 entries (of 51200):
    indirect_reference_table.cc:256]     51199: 0x13640000 java.lang.Thread
    indirect_reference_table.cc:256]     51198: 0x13600000 java.lang.Thread
    indirect_reference_table.cc:256]     51197: 0x135c0000 java.lang.Thread
    indirect_reference_table.cc:256]     51196: 0x13580000 java.lang.Thread
    indirect_reference_table.cc:256]     51195: 0x13540000 java.lang.Thread
    indirect_reference_table.cc:256]     51194: 0x13500000 java.lang.Thread
    indirect_reference_table.cc:256]     51193: 0x134c0000 java.lang.Thread
    indirect_reference_table.cc:256]     51192: 0x13480000 java.lang.Thread
    indirect_reference_table.cc:256]     51191: 0x13400000 java.lang.Thread
    indirect_reference_table.cc:256]     51190: 0x13440000 java.lang.Thread
    indirect_reference_table.cc:256]   Summary:
    indirect_reference_table.cc:256]      9779 of java.lang.Class (9779 unique instances)
    indirect_reference_table.cc:256]        64 of java.lang.Thread (64 unique instances)
    indirect_reference_table.cc:256]        47 of java.lang.DexCache (47 unique instances)
    indirect_reference_table.cc:256]        12 of dalvik.system.PathClassLoader (6 unique instances)
    indirect_reference_table.cc:256]         3 of aeP (1 unique instances)
    indirect_reference_table.cc:256]         2 of java.lang.ThreadGroup (2 unique instances)
    indirect_reference_table.cc:256]         1 of com.google.android.gms.internal.zzcki
    indirect_reference_table.cc:256]         1 of android.opengl.GLSurfaceView
    indirect_reference_table.cc:256]         1 of com.mapbox.mapboxsdk.maps.NativeMapView
    indirect_reference_table.cc:256]         1 of com.mapbox.mapboxsdk.maps.MapView$5
    indirect_reference_table.cc:256]         1 of com.andreucci.andreuccicodrive.adapters.TomTomRouteAdapter
    indirect_reference_table.cc:256]         1 of org.chromium.android_webview.AwSettings
    indirect_reference_table.cc:256]         1 of org.chromium.android_webview.AwContentsClientBridge
    indirect_reference_table.cc:256]         1 of org.chromium.android_webview.AwAutofillClient
    indirect_reference_table.cc:256]         1 of org.chromium.android_webview.AwContents
    indirect_reference_table.cc:256]         1 of org.chromium.content.browser.input.ImeAdapterImpl
    indirect_reference_table.cc:256]         1 of org.chromium.content.browser.GestureListenerManagerImpl
    indirect_reference_table.cc:256]         1 of org.chromium.content.browser.accessibility.captioning.CaptioningController
    indirect_reference_table.cc:256]         1 of agq
    indirect_reference_table.cc:256]         1 of org.chromium.content.browser.ContentUiEventHandler
    indirect_reference_table.cc:256]         1 of org.chromium.content.browser.TapDisambiguator
    indirect_reference_table.cc:256]         1 of org.chromium.content.browser.input.TextSuggestionHost
    indirect_reference_table.cc:256]         1 of org.chromium.content.browser.selection.SelectionPopupControllerImpl
    indirect_reference_table.cc:256]         1 of org.chromium.android_webview.AwGLFunctor
    indirect_reference_table.cc:256]         1 of aeN
    indirect_reference_table.cc:256]         1 of aec
    indirect_reference_table.cc:256]         1 of com.google.android.gms.ads.internal.webview.ab
    indirect_reference_table.cc:256]         1 of org.chromium.android_webview.AwTracingController
    indirect_reference_table.cc:256]         1 of org.chromium.android_webview.AwQuotaManagerBridge
    indirect_reference_table.cc:256]         1 of org.chromium.content.browser.ContentViewCoreImpl
    indirect_reference_table.cc:256]         1 of org.chromium.content.browser.JavascriptInjectorImpl
    indirect_reference_table.cc:256]         1 of agn
    indirect_reference_table.cc:256]         1 of org.chromium.content.browser.input.SelectPopup
    indirect_reference_table.cc:256]         1 of org.chromium.content.browser.selection.SmartSelectionClient
    indirect_reference_table.cc:256]         1 of ad
    indirect_reference_table.cc:256]         1 of com.google.android.gms.dynamite.zzh
    indirect_reference_table.cc:256]         1 of java.lang.BootClassLoader
    indirect_reference_table.cc:256]         1 of java.util.HashSet
    indirect_reference_table.cc:256]         1 of android.net.ConnectivityThread
    indirect_reference_table.cc:256] `

**Comment by tobrun** [comment]
this is fixed with the jni 0.4.0, aimed to ship with 6.6.0
